modAionImpl/src/org/aion/zero/impl/blockchain/AionPendingStateImpl.java

```diff
@@ -398,15 +471,20 @@ private boolean addPendingTransactionImpl(final AionTransaction tx, BigInteger t
                 LOG.trace("addPendingTransactionImpl: [{}]", tx.toString());
             }
 
-            AionTransaction rtn = this.txPool.add(tx);
-            if (rtn != null && !rtn.equals(tx)) {
-                AionTxReceipt rp = new AionTxReceipt();
-                rp.setTransaction(rtn);
-                receivedTxs.remove(ByteArrayWrapper.wrap(rtn.getHash()));
-                fireTxUpdate(rp, PendingTransactionState.DROPPED, getBestBlock());
+            if (bufferEnable) {
+                txBuffer.add(txSum);
```

Comment:
This may get expensive if we update on every transaction

Reply:
you mean the CopyOnWriteArrayList? It can be refactored later.

Reply:
replace to the Collections.synchronizedList (new ArrayList<>() )